app/src/main/java/com/toygar/tasks/adapters/TaskListAdapter.kt

```diff
@@ -0,0 +1,72 @@
+package com.toygar.tasks.adapters
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.toygar.tasks.databinding.TaskCardBinding
+import com.toygar.tasks.models.Task
+import com.toygar.tasks.util.TaskDiffCallback
+import java.lang.reflect.Field
+
+class TaskListAdapter(
+    context: Context,
+    private var taskList: List<Task>,
+    private val deleteMethod: (Task) -> Unit
+): RecyclerView.Adapter<TaskListAdapter.TaskViewHolder>() {
+
+    private val inflater: LayoutInflater = LayoutInflater.from(context)
+    private var comparator: Comparator<Task> = compareBy(Task::dueDate)
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): TaskViewHolder {
+
+        val itemBinding : TaskCardBinding = TaskCardBinding.inflate(inflater, parent, false)
+        return TaskViewHolder(itemBinding)
+    }
+
+    override fun onBindViewHolder(holder: TaskViewHolder, position: Int) {
+        val current : Task = taskList[position]
+        holder.bind(current, deleteMethod)
+    }
+
+    override fun getItemCount(): Int = taskList.size
+
+    fun setData(newTasks: List<Task>) {
+        val newSortedTasks = newTasks.sortedWith(comparator)
+        val diffCallback = TaskDiffCallback(taskList, newSortedTasks)
+        val diffResult = DiffUtil.calculateDiff(diffCallback)
+        taskList = newSortedTasks
+        diffResult.dispatchUpdatesTo(this)
+    }
+
+    fun sortData(sortField: String) {
+        val hm = hashMapOf<String, Comparator<Task>>()
+        hm.put("Description", compareBy(Task::description))
+        hm.put("Name", compareBy(Task::name))
+        hm.put("Due Date", compareBy(Task::dueDate))
+        hm.put("Priority", compareBy(Task::priority))
+        hm.withDefault { compareBy(Task::dueDate) }
+
+        comparator = hm[sortField]!!
+
+        setData(taskList)
+    }
+
+    class TaskViewHolder(
+        private val itemBinding: TaskCardBinding
```

Comment:
ViewHolder içinde binding objeleri tutmak yerine bind() method'unda şu şekilde binding objesine erişebilirsin:

`val binding = DataBindingUtil.bind(itemView)`

ViewHolder içinde `itemView` değişkeni oluyor root view'a erişmek için.

Reply:
Bindinge böyle erişsem bile, hala super classın constructorına vermek için itemBinding parametresi almam gerekiyor constructorda

Reply:
22. satırda Ağabey'in dediği şekilde yapılabilir. Böyle kalmasında da sakınca yok gibi.